airqo-platform/AirQo-api
# verify email address during request access

# verify email address during request access

**_WHAT DOES THIS PR DO?_**
verify email address during request access. A lot of different factors can go into verifying an email address, some common issues this PR addresses:

- Non-existent email addresses
- Temporary email address
- Common typos in the email address domain name like xxxx@gmaill.com, xxxx@gmaiil.com

**_WHAT JIRA STORY/TASK IS RELATED TO THIS PR?_**
N/A

**_WHAT IS THE LINK TO THE PR BRANCH_**
https://github.com/airqo-platform/AirQo-api/tree/en-email-validation

**_HOW DO I TEST OUT THIS PR?_**
Please note that the latest `.env `has been shared in the [engineering Slack channel](https://airqo.slack.com/archives/C0278FK2EGG/p1636186085000500).
```
cd src/auth-service
npm install
npm run stage-mac
```

**_WHICH ENDPOINTS SHOULD BE READY FOR TESTING?:_** 
Attempt to request access using a wrong email address, one that has a typo in it. Example: martin@gmailll.com
[Documentation of request access endpoint](https://app.gitbook.com/o/-MCogOVGQqjUharJKchL/s/-MKqyQkcfs54GYi-q96G/auth-service/request-access#request-access)

_request access:_ 
[POST] http://localhost:3000/api/v1/users/candidates/register?tenant=airqo



**_IS THERE ANY JENKINS CONSOLE LINK FOR CODE COVERAGE AND BUILD INFO?_**
N/A

**_ARE THERE ANY RELATED PRs?_**
N/A

**_SCREENSHOTS?_**
<img width="925" alt="Screenshot 2021-11-06 at 11 09 27" src="https://user-images.githubusercontent.com/1590213/140602906-32b7b770-9e5e-43ec-8521-da4142cb6559.png">
